src-srv/routes.ts

```diff
@@ -0,0 +1,227 @@
+import fs from 'node:fs'
+import path from 'node:path'
+import type { Request, Response } from 'express'
+import type { Application, WebsocketRequestHandler } from 'express-ws'
+import type { Repository } from 'utils/Repository.ts'
+
+/* Route types */
+interface Route {
+  path: string
+  params: string[]
+  handlers?: RouteHandlers
+}
+
+interface RouteInitContext {
+  repository: Repository
+  [key: string]: unknown
+}
+
+interface RouteContext extends RouteInitContext {
+  res: Response
+}
+
+interface RouteContentResponse {
+  payload: unknown
+  statusCode?: number
+}
+
+interface RouteStatusResponse {
+  statusCode: number
+  statusMessage: string
+}
+
+export type RouteHandler = (req: Request, context: RouteContext) => Promise<RouteContentResponse | RouteStatusResponse>
+
+export interface ApiResponseInterface {
+  isStatus: (value: unknown) => value is RouteStatusResponse
+  isContent: (value: unknown) => value is RouteContentResponse
+}
+
+interface RouteHandlers {
+  GET?: RouteHandler
+  POST?: RouteHandler
+  PUT?: RouteHandler
+  PATCH?: RouteHandler
+  DELETE?: RouteHandler
+  WEB_SOCKET?: WebsocketRequestHandler
+}
+
+type RouteMap = Record<string, Route>
+
+export const ApiResponse: ApiResponseInterface = {
+  isStatus: (value): value is RouteStatusResponse => {
+    return typeof value === 'object' && ('statusCode' in value && 'statusMessage' in value)
+  },
+  isContent: (value): value is RouteContentResponse => {
+    return typeof value === 'object' && 'payload' in value
+  }
+}
+
+/*
+ * Map directory structure under api/ to express route paths.
+ * Direcory or file names enclosed in brackets, i.e api/planning[id]/[action].ts
```

Comment:
correction => "Directory or file names enclosed in brackets, i.e api/planning/[id]/[action].ts"